### PR TITLE
[V3 Launcher] Reinstall Red option

### PR DESCRIPTION
**Pull request by laggron42**
### Type

- [ ] Bugfix
- [ ] Enhancement
- [x] New feature

### Description of the changes

This PR adds a new button in the launcher: `Reinstall Red`. This is similar to the v2 option, it can reinstall requirements and discard code changes, reset data, or both. A lit of things were edited in `launcher.py` and `setup.py` to make it (mostly some features moved to a function so they can be used twice without repeating).
**That is not tested yet** because of the packages conflict with `websockets` which makes the update glitching. Please test it and push any changes if you're a maintainer so we can merge this before beta 12.

### CODE REVIEW

**Review by Kowlin — DISMISSED** [review]
Changes required since this is removing functionality from another PR.

File: redbot/setup.py [hunk 1]
```diff
@@ -287,7 +283,7 @@ def basic_setup():
                 if confirm("Would you like to import your data? (y/n) "):
                     await json_to_mongo(current_data_dir, storage_details)
         else:
-            storage_details = instance_data["STORAGE_DETAILS"]
+            storage_details = default_dirs["STORAGE_DETAILS"]
```
Comment (palmtree5) [inline redbot/setup.py]: NO NO NO. This should NOT have been changed. `storage_details` is set to the storage details from the instance settings for a very, very good reason (the actual storage details are needed if the user wants to import from MongoDB to JSON)

**Review by palmtree5 — DISMISSED** [review]

File: redbot/setup.py (hunk 1 above)
Comment (laggron42) [reply redbot/setup.py]: That wasn't intended. Seems that's from another PR. Removing that...

**Comment by laggron42** [comment]
Fixed wrong changes. Needs testing

**Comment by laggron42** [comment]
I recorded a video showing the features on the last commit
https://youtu.be/L4rf_ejv4xE

**Review by Kowlin — APPROVED** [review]

**Pull request merged**